api.lua

```diff
@@ -3,17 +3,48 @@
 function areas:getAreasAtPos(pos)
 	local a = {}
 	local px, py, pz = pos.x, pos.y, pos.z
-	for id, area in pairs(self.areas) do
-		local ap1, ap2 = area.pos1, area.pos2
-		if px >= ap1.x and px <= ap2.x and
-		   py >= ap1.y and py <= ap2.y and
-		   pz >= ap1.z and pz <= ap2.z then
-			a[id] = area
+	if self.store then
+		local areas = self.store:get_areas_for_pos(pos, false, true)
+		for store_id, store_area in pairs(areas) do
+			local id = tonumber(store_area.data)
+			a[id] = self.areas[id]
+		end
+	else
+		for id, area in pairs(self.areas) do
+			local ap1, ap2 = area.pos1, area.pos2
+			if px >= ap1.x and px <= ap2.x and
+			py >= ap1.y and py <= ap2.y and
+			pz >= ap1.z and pz <= ap2.z then
+				a[id] = area
+			end
 		end
 	end
 	return a
 end
 
+function areas:getAreasForArea(pos1, pos2)
+	local res = {}
+	if self.store then
+		local areas = self.store:get_areas_in_area(pos1,
+			pos2, true, false, true)
+		for store_id, store_area in pairs(areas) do
+			local id = tonumber(store_area.data)
+			res[id] = self.areas[id]
+		end
+	else
+		for id, area in pairs(self.areas) do
+			local p1, p2 = area.pos1, area.pos2
+			if (p1.x <= pos2.x and p2.x >= pos1.x) and
+					(p1.y <= pos2.y and p2.y >= pos1.y) and
+					(p1.z <= pos2.z and p2.z >= pos1.z) then
```

Comment:
This actually checks if the passed area is entirely contained by the area you're iterating over, but we're interested in any areas that share even one node with the passed area.